tests/models/timm_wrapper/test_modeling_timm_wrapper.py

```diff
@@ -0,0 +1,364 @@
+    @slow
+    def test_inference_image_classification_head(self):
+        checkpoint = "timm/resnet18.a1_in1k"
+        model = TimmWrapperForImageClassification.from_pretrained(checkpoint, device_map=torch_device).eval()
+        image_processor = TimmWrapperImageProcessor.from_pretrained(checkpoint)
+
+        image = prepare_img()
+        inputs = image_processor(images=image, return_tensors="pt").to(torch_device)
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs)
+
+        # verify the shape and logits
+        expected_shape = torch.Size((1, 1000))
+        self.assertEqual(outputs.logits.shape, expected_shape)
+
+        expected_label = 281  # tabby cat
+        self.assertEqual(torch.argmax(outputs.logits).item(), expected_label)
+
+        expected_slice = torch.tensor([-11.2618, -9.6192, -10.3205]).to(torch_device)
+        resulted_slice = outputs.logits[0, :3]
+        is_close = torch.allclose(resulted_slice, expected_slice, atol=1e-4)
+        self.assertTrue(is_close, f"Expected {expected_slice}, but got {resulted_slice}")
+
+    @slow
+    @require_bitsandbytes
+    def test_inference_image_classification_quantized(self):
+        from transformers import BitsAndBytesConfig
+
+        checkpoint = "timm/vit_small_patch16_384.augreg_in21k_ft_in1k"
+
+        quantization_config = BitsAndBytesConfig(load_in_8bit=True)
+        model = TimmWrapperForImageClassification.from_pretrained(
+            checkpoint, quantization_config=quantization_config, device_map=torch_device
+        ).eval()
+        image_processor = TimmWrapperImageProcessor.from_pretrained(checkpoint)
+
+        image = prepare_img()
+        inputs = image_processor(images=image, return_tensors="pt").to(torch_device)
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs)
+
+        # verify the shape and logits
+        expected_shape = torch.Size((1, 1000))
+        self.assertEqual(outputs.logits.shape, expected_shape)
+
+        expected_label = 281  # tabby cat
+        self.assertEqual(torch.argmax(outputs.logits).item(), expected_label)
+
+        expected_slice = torch.tensor([-2.4043, 1.4492, -0.5127]).to(outputs.logits.dtype)
+        resulted_slice = outputs.logits[0, :3].cpu()
+        is_close = torch.allclose(resulted_slice, expected_slice, atol=1e-3)
+        self.assertTrue(is_close, f"Expected {expected_slice}, but got {resulted_slice}")
+
+    @slow
+    def test_transformers_model_equivalent_to_timm(self):
+        # check that wrapper logits are the same as timm model logits
+
+        # some popular ones
+        model_names = [
+            "vit_small_patch16_384.augreg_in21k_ft_in1k",
+            "resnet50.a1_in1k",
+            "tf_mobilenetv3_large_minimal_100.in1k",
+            "swin_tiny_patch4_window7_224.ms_in1k",
+            "ese_vovnet19b_dw.ra_in1k",
+            "hrnet_w18.ms_aug_in1k",
+        ]
+        image = prepare_img()
+
```

Comment:
Looks like these tests are focused on models with classification heads with a classification wrapper...

Other scenarios to consider
* timm model w/ classification head used with TimmWrapperModel
* timm model w/o classification head use with TimmWrapperModel
* timm model w/o classification head use with TimmWrapperForImageClassification  (head needs to be added by wrapper)

Some models wihout heads to check
* vit_base_patch16_siglip_224.webli ... this is OpenCLIP model w/o timm config in hub though https://huggingface.co/timm/ViT-B-16-SigLIP 
* vit_small_patch16_224.dino

Reply:
The `TimmWrapperModel` represents a model without a classification head, while the `TimmWrapperForImageClassification` includes a classification head. The wrapper itself does not add any heads, instead, the classification head is provided entirely by the original `timm` model. Such a behavior indeed is different from other `transformers` models, however, it allows load weights to classification head with no issues + make weights timm-backward-compatible.